### PR TITLE
Improve pipe for general use

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

This PR generalizes the file pipe mechanism that was developed for FL Hub initially.

- Improved file name format
- Split name processing functions into reusable utils
- Introduced FileAccessor to allow the file content to be written/read with different solutions (e.g. H5)
- Updated all HUB code accordingly

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_common/hub/hub_controller.py [hunk 1]
```diff
@@ -348,39 +350,45 @@ def _control_flow(self, abort_signal: Signal, fl_ctx: FLContext):
                     )
                     return
             else:
-                if topic in [Topic.ABORT, Topic.END, Topic.PEER_GONE]:
+                assert isinstance(msg, Message)
```
Comment (YuanTingHsieh) [inline nvflare/app_common/hub/hub_controller.py:353]: note that assert will not work if start python with "-O" options (which should be the case for production use)

File: nvflare/fuel/utils/pipe/file_pipe.py [hunk 2]
```diff
@@ -85,30 +108,25 @@ def _clear_dir(p: str):
...
             raise BrokenPipeError("pipe broken")
         try:
-            with open(tmp_path, "wb") as f:
-                f.write(data_bytes)
+            self.accessor.write(msg.data, tmp_path)
             os.rename(tmp_path, file_path)
         except FileNotFoundError:
             raise BrokenPipeError("pipe closed")
+
+        print(f"********** created file: {file_path}")
```
Comment (YuanTingHsieh) [inline nvflare/fuel/utils/pipe/file_pipe.py:129]: Should we remove this print?

File: nvflare/fuel/utils/pipe/file_name_utils.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if num_parts != 3:
+            raise ValueError(f"bad file name for request: {file_name} - must be 3 parts but got {num_parts}")
+        return Message.new_request(topic, data, msg_id)
+    elif msg_type == Message.REPLY:
+        if num_parts != 4:
+            raise ValueError(f"bad file name for request: {file_name} - must be 4 parts but got {num_parts}")
+        req_id = parts[2]
+        return Message.new_reply(topic, data, req_id, msg_id)
+    else:
+        raise ValueError(f"bad file name for request: {file_name} - invalid msg type '{msg_id}'")
```
Comment (YuanTingHsieh) [inline nvflare/fuel/utils/pipe/file_name_utils.py:62]: msg_id should be msg_type ?

**Review by YuanTingHsieh — COMMENTED** [review]
Logic LGTM, just minor comments.

File: nvflare/app_common/hub/hub_controller.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/app_common/hub/hub_controller.py]: Removed as suggested

File: nvflare/fuel/utils/pipe/file_pipe.py (hunk 2 above)
Comment (yanchengnv) [reply nvflare/fuel/utils/pipe/file_pipe.py]: Removed.

File: nvflare/fuel/utils/pipe/file_name_utils.py (hunk 3 above)
Comment (yanchengnv) [reply nvflare/fuel/utils/pipe/file_name_utils.py]: Good catch! Fixed.

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM

**Pull request merged**